src/summarycode/summarizer.py

```diff
@@ -502,11 +503,29 @@ def package_summarizer(resource, children, keep_details=False):
         logger_debug('package_summarizer: for:', resource,
                      'packages are:', packs)
 
+    package_purls = []
+    for package in packages:        
+        if package.get("purl", False):
+            package_purls.append(package["purl"])
+
     # Collect direct children packages summary
     for child in children:
         child_summaries = get_resource_summary(child, key='packages', as_attribute=False) or []
-        packages.extend(child_summaries)
-
+        for child_summary in child_summaries:
+            values = [child_summary['value']] * child_summary['count']
```

Comment:
I am a tad puzzled by this line... and I am not sure I get what it does... can you explain a bit? Why create a list with duplicated values like that?

Reply:
@pombredanne Sorry for the delayed response.

I was trying to understand how to collect `package_urls` and count in order to make sure it works as intended. ( ref. @JonoYang 's [comment](https://github.com/nexB/scancode-toolkit/issues/1759#issuecomment-566812536)).

For that reason, I was trying a couple of Ideas :
1. Was this [addition](https://github.com/nexB/scancode-toolkit/pull/1864/files#diff-52211dc043d1ddb236472d7177e24fcaR506)
2. Was to utilize the [existing block](https://github.com/nexB/scancode-toolkit/blob/develop/src/summarycode/summarizer.py#L506) similar to the way`license_summarizer` method does it.

Funny enough, I missed out to remove the other and didn't realize it until you pointed it out.

I'll remove it, Thanks for pointing it out. Please feel free to correct me if I'm wrong.

Thanks!